features/nvstore/source/nvstore.h

```diff
@@ -394,14 +422,14 @@ class NVStore : private mbed::NonCopyable<NVStore> {
     /**
      * @brief Actual logics of set API (covers also set_once and remove APIs).
      *
-     * @param[in]  key                    key.
+     * @param[out] key                    key (both input and output).
      * @param[in]  buf_size               Buffer size (bytes).
      * @param[in]  buf                    Input Buffer.
      * @param[in]  flags                  Record flags.
      *
      * @returns 0 for success, nonzero for failure.
      */
-    int do_set(uint16_t key, uint16_t buf_size, const void *buf, uint16_t flags);
+    int do_set(uint16_t &key, uint16_t buf_size, const void *buf, uint16_t flags);
```

Comment:
Have you considered exposing `do_set` as a public overload of the `set` function? Chosing flags would gives more options to the user.

Reply:
Don't think it's the way to go here. Aim is to keep the API as simple as possible (as well as keeping the user on a short leash).